app/questionnaire/value_source_resolver.py

```diff
@@ -127,13 +128,25 @@ def _resolve_calculated_summary_value_source(
         """
         calculated_summary_block: Mapping[str, Any] = self.schema.get_block(value_source["identifier"])  # type: ignore
         calculation = calculated_summary_block["calculation"]
-        operator = self.get_calculation_operator(calculation["calculation_type"])
-        list_item_id = self._resolve_list_item_id_for_value_source(value_source)
-        values = [
-            self._get_answer_value(answer_id=answer_id, list_item_id=list_item_id)
-            for answer_id in calculation["answers_to_calculate"]
-        ]
-        return operator([value for value in values if value])  # type: ignore
+        if calculation.get("answers_to_calculate"):
+            operator = self.get_calculation_operator(calculation["calculation_type"])
+            list_item_id = self._resolve_list_item_id_for_value_source(value_source)
+            values = [
+                self._get_answer_value(answer_id=answer_id, list_item_id=list_item_id)
+                for answer_id in calculation["answers_to_calculate"]
+            ]
+            return operator([value for value in values if value])  # type: ignore
+
+        evaluate_calculated_summary = rule_evaluator.RuleEvaluator(
+            self.schema,
+            self.answer_store,
+            self.list_store,
+            self.metadata,
+            self.response_metadata,
+            location=self.location,
+        )
+
+        return evaluate_calculated_summary.evaluate(calculation["operation"])  # type: ignore
```

Comment:
I understand why this is done but it highlights a problem in design. We now have a RE > VSR > RE circular dependency. I suspect that why it has even been imported entirely?.

In a way it does give us the flexility to just pass to rule evaluator, but a bit of a code smell. Happy to be pragmatic and look at this when we have some time. In future or when it becomes a problem. It might become one when you do the answer only on path fix.